src/main/java/randoop/generation/DemandDrivenInputCreation.java

```diff
@@ -0,0 +1,489 @@
+    workList.add(initType);
+    while (!workList.isEmpty()) {
+      Type currentType = workList.poll();
+
+      // Log the unspecified classes that are used in demand-driven input creation.
+      if (!SPECIFIED_CLASSES.contains(currentType.getRuntimeClass().getName())) {
+        unspecifiedClasses.add(currentType.getRuntimeClass());
+      }
+
+      // Only consider the type if it is not a primitive type or if it hasn't already been
+      // processed.
+      if (!processed.contains(currentType) && !currentType.isNonreceiverType()) {
+        Class<?> currentClass = currentType.getRuntimeClass();
+        List<Executable> executableList = new ArrayList<>();
+
+        // Adding constructors if the current type is what we are looking for.
+        if (t.equals(currentType)) {
+          for (Constructor<?> constructor : currentClass.getConstructors()) {
+            executableList.add(constructor);
+          }
+        }
+
+        // Adding methods that return the current type.
+        for (Method method : currentClass.getMethods()) {
+          executableList.add(method);
+        }
+
+        // The first call checks for methods that return the specified type. Subsequent calls
+        // check for methods that return the current type.
+        Type returnType = initialCall ? t : currentType;
+        for (Executable executable : executableList) {
+          if (executable instanceof Constructor
+              || (executable instanceof Method
+                  && ((Method) executable).getReturnType().equals(returnType.getRuntimeClass()))) {
+
+            // Obtain the input types and output type of the executable.
+            List<Type> inputTypeList = classArrayToTypeList(executable.getParameterTypes());
+            // If the executable is a non-static method, add the receiver type to
+            // the front of the input type list.
+            if (executable instanceof Method && !Modifier.isStatic(executable.getModifiers())) {
+              inputTypeList.add(0, new NonParameterizedType(currentClass));
+            }
+            TypeTuple inputTypes = new TypeTuple(inputTypeList);
+            CallableOperation callableOperation =
+                executable instanceof Constructor
+                    ? new ConstructorCall((Constructor<?>) executable)
+                    : new MethodCall((Method) executable);
+            NonParameterizedType declaringType = new NonParameterizedType(currentClass);
+            TypedOperation typedClassOperation =
+                new TypedClassOperation(callableOperation, declaringType, inputTypes, returnType);
+
+            // Add the method call to the producerMethods.
+            producerMethodsList.add(typedClassOperation);
+            producerParameterTypes.addAll(inputTypeList);
+          }
+          processed.add(currentType);
+          // Recursively search for methods that construct objects of the specified type.
+          workList.addAll(producerParameterTypes);
```

Comment:
This is looking for methods with return type `t`, whereas I would have expected a search for methods returning a type in `producerParameterTypes` for a recursive search.